api/src/main/java/io/opencensus/resource/Resource.java

```diff
@@ -99,14 +104,34 @@ public static Resource create() {
    *     ASCII string or exceed {@link #MAX_LENGTH} characters.
    * @since 0.18
    */
-  public static Resource create(@Nullable String type, Map<String, String> labels) {
+  // TODO(mayurkale): Use this method while creating resource information from GCP, AWS etc.
+  // Counterpart :
+  // https://github.com/census-ecosystem/opencensus-go-resource/blob/master/aws/aws.go#L35-L41
+  public static Resource create(/*@Nullable*/ String type, Map<String, String> labels) {
     return createInternal(
         type,
         Collections.unmodifiableMap(
             new LinkedHashMap<String, String>(Utils.checkNotNull(labels, "labels"))));
   }
 
-  private static Resource createInternal(@Nullable String type, Map<String, String> labels) {
+  /**
+   * Returns a {@link Resource} that runs all input resources sequentially and merges their results.
+   * In case a type of label key is already set, the first set value takes precedence.
+   *
+   * @param resources a list of resources.
+   * @return a {@code Resource}.
+   * @since 0.18
+   */
+  @javax.annotation.Nullable
+  public Resource multiDetector(List</*@Nullable*/ Resource> resources) {
```

Comment:
Also why not have this as static method?

Reply:
Done